core/src/scene/importer.cpp

```diff
@@ -82,140 +69,89 @@ Node Importer::applySceneImports(const std::string& scenePath, const std::string
     Node root = Node();
 
     LOGD("Processing scene import Stack:");
-    std::vector<std::string> sceneStack;
-    importScenes(root, fullPath, sceneStack);
+    std::vector<Url> sceneStack;
+    importScenesRecursive(root, rootScenePath, sceneStack);
 
     return root;
 }
 
-void Importer::processScene(const std::string &scenePath, const std::string &sceneString) {
+void Importer::processScene(const Url& scenePath, const std::string &sceneString) {
 
-    LOGD("Process: '%s'", scenePath.c_str());
+    LOGD("Process: '%s'", scenePath.string().c_str());
+
+    // Don't load imports twice
+    if (m_scenes.find(scenePath) != m_scenes.end()) {
```

Comment:
Good point! it should be redundant. At least we should have a better solution :) 

The problem is that the same download path can be enqueued more than once. The first finished download adds the processed scene node. This only solves that the second yaml gets discarded.